src/arizona_parse_transform.erl

```diff
@@ -186,42 +385,674 @@ transform_stateless_to_ast(StatelessList) when is_list(StatelessList) ->
+
+%% Build mapping of function specs to their variable-to-binding mappings
+-spec build_function_bindings_map(AbstractSyntaxTrees, ArizonaFunctions) -> FunctionBindings when
+    AbstractSyntaxTrees :: [erl_parse:abstract_form()],
+    ArizonaFunctions :: [function_spec()],
+    FunctionBindings :: #{function_spec() => #{binary() => binary()}}.
+build_function_bindings_map(AbstractSyntaxTrees, ArizonaFunctions) ->
+    lists:foldl(
+        fun(Form, Acc) ->
+            case Form of
+                {function, _Line, FuncName, Arity, _Clauses} = Function ->
+                    FuncSpec = {FuncName, Arity},
+                    case lists:member(FuncSpec, ArizonaFunctions) of
+                        true ->
+                            Bindings = analyze_function_for_bindings(Function),
+                            Acc#{FuncSpec => Bindings};
+                        false ->
+                            Acc
+                    end;
+                _ ->
+                    Acc
+            end
+        end,
+        #{},
+        AbstractSyntaxTrees
+    ).
+
+%% Analyze function AST to extract arizona_socket:get_binding calls
+-spec analyze_function_for_bindings(Function) -> VarBindings when
+    Function :: erl_parse:abstract_form(),
+    VarBindings :: #{binary() => [binary()]}.
+analyze_function_for_bindings({function, _Line, _Name, _Arity, _Clauses} = Function) ->
+    %% Use erl_syntax_lib:map to traverse ALL AST nodes systematically
+    CollectedBindings = collect_bindings_from_function(Function),
+    %% Resolve transitive dependencies
+    resolve_transitive_dependencies(CollectedBindings);
+analyze_function_for_bindings(_) ->
+    #{}.
+
+%% Collect all arizona_socket:get_binding calls from function using erl_syntax traversal
+-spec collect_bindings_from_function(Function) -> Bindings when
+    Function :: erl_parse:abstract_form(),
+    Bindings :: #{binary() => binary()}.
+collect_bindings_from_function(Function) ->
+    erl_syntax_lib:fold(
+        fun(Node, Acc) ->
+            analyze_node_for_bindings(erl_syntax:revert(Node), Acc)
+        end,
+        #{},
+        Function
+    ).
+
+%% Analyze individual AST node for arizona_socket:get_binding patterns
+-spec analyze_node_for_bindings(AstNode, Acc) -> NewAcc when
+    AstNode :: erl_parse:abstract_expr(),
+    Acc :: #{binary() => binary() | [binary()]},
+    NewAcc :: #{binary() => binary() | [binary()]}.
+analyze_node_for_bindings(AstNode, Acc) ->
+    case AstNode of
+        %% Match: Var = RHS (any assignment)
+        {match, _Line, {var, _, VarName}, RHS} ->
+            %% Collect all bindings this variable depends on (direct + transitive)
+            BindingDependencies = collect_binding_dependencies(RHS),
+            VariableDependencies = collect_variable_dependencies(RHS),
+            AllDependencies = lists:usort(BindingDependencies ++ VariableDependencies),
+            case AllDependencies of
```

Comment:
Maybe best to have as a separate function (?)

Reply:
https://github.com/arizona-framework/arizona/pull/329/commits/bd7e7b6585a6114f94d2a8b76ab313a225b75cd6